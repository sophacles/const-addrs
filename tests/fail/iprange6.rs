use const_addrs::iprange6;

fn main() {
    let _ = iprange6!("2001:db8::32:23", "2001:db8::32::ffff");
    let _ = iprange6!("2001:db8::32::23", "2001:db8::32:ffff");
    let _ = iprange6!("2001:db8::32::23", "2001:db8::32::ffff");
    let _ = iprange6!("2001:db8::32:23", "10.0.0.2");
    let _ = iprange6!("10.0.0.1", "2001:db8::32:23");
}
