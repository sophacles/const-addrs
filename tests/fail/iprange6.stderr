error: invalid IP address syntax
 --> tests/fail/iprange6.rs:4:42
  |
4 |     let _ = iprange6!("2001:db8::32:23", "2001:db8::32::ffff");
  |                                          ^^^^^^^^^^^^^^^^^^^^

error: invalid IP address syntax
 --> tests/fail/iprange6.rs:5:23
  |
5 |     let _ = iprange6!("2001:db8::32::23", "2001:db8::32:ffff");
  |                       ^^^^^^^^^^^^^^^^^^

error: macro expansion ignores `compile_error` and any tokens following
 --> tests/fail/iprange6.rs:6:43
  |
6 |     let _ = iprange6!("2001:db8::32::23", "2001:db8::32::ffff");
  |             ------------------------------^^^^^^^^^^^^^^^^^^^^- caused by the macro expansion here
  |
  = note: the usage of `iprange6!` is likely invalid in expression context

error: invalid IP address syntax
 --> tests/fail/iprange6.rs:6:23
  |
6 |     let _ = iprange6!("2001:db8::32::23", "2001:db8::32::ffff");
  |                       ^^^^^^^^^^^^^^^^^^

error: Addr is IPv4 but IPv6 is expected
 --> tests/fail/iprange6.rs:7:42
  |
7 |     let _ = iprange6!("2001:db8::32:23", "10.0.0.2");
  |                                          ^^^^^^^^^^

error: Addr is IPv4 but IPv6 is expected
 --> tests/fail/iprange6.rs:8:23
  |
8 |     let _ = iprange6!("10.0.0.1", "2001:db8::32:23");
  |                       ^^^^^^^^^^
